src/App.js

```diff
@@ -525,6 +537,89 @@ export const App = () => {
 		}
 	};
 
+	const onDeposit = async (neoBalanceTemp) => {
+		if (depositQuantity * 1e8 >= 1 && depositQuantity * 1e8 <= neoBalanceTemp) {
+			onModal('approveRequest');
+			const invocations = [{
+				scriptHash: NeoFSContract.gasToken,
+				operation: 'transfer',
+				args: [
+					{ type: 'Hash160', value: Neon.create.account(walletData.account.address).scriptHash },
+					{ type: 'Hash160', value: Neon.create.account(NeoFSContract.account).scriptHash },
+					{ type: 'Integer', value: depositQuantity * 1e8 },
+					{ type: 'ByteArray', value: '' },
+				]
+			}];
+
+			const signers = [{
+				scopes: 'CalledByEntry',
+				account: Neon.create.account(walletData.account.address).scriptHash,
+			}];
+
+			let response = '';
+			if (walletData.name === 'o3-desktop') {
+				response = await neo3Dapi.invoke({ ...invocations[0], signers }).catch((err) => handleError(err));
+			} else if (neolineN3) {
+				response = await neolineN3.invoke({ ...invocations[0], signers }).catch((err) => handleError(err));
+			} else if (dapi) {
+				response = await dapi.invoke({ ...invocations[0], signers }).catch((err) => handleError(err));
+			} else {
+				response = await wcSdk.invokeFunction({ invocations, signers }).catch((error) => {
+					if (error.message === 'Failed or Rejected Request') {
+						onModal('failed', 'Failed or Rejected Request');
+					} else if (error.message === 'Error: intrinsic gas too low') {
+						onModal('failed', 'Transaction intrinsic gas too low');
+					} else {
+						onModal('failed', 'Something went wrong, try again');
+					}
+				});
+			}
+			if (!response.error) {
+				setDepositQuantity(0);
+				onModal('success', response.txid ? response.txid : response);
+			}
+		} else {
+			onPopup('failed', 'Incorrect quantity value');
+		}
+	};
+
+	const onWithdraw = async (neoFSBalanceTemp) => {
+		if (withdrawQuantity >= 1 && withdrawQuantity * 1e12 <= neoFSBalanceTemp) {
```

Comment:
Because this one can't, we need some integer amount of GAS.

Reply:
in this case, it can't, because the input with onKeyPress function, which allows only integer numbers